src/lib/parameters/parameters.cpp

```diff
@@ -808,32 +766,19 @@ param_set_no_notification(param_t param, const void *val)
 bool
 param_used(param_t param)
 {
-	int param_index = param_get_index(param);
-
-	if (param_index < 0) {
-		return false;
+	if (handle_in_range(param)) {
+		return params_active[param];
 	}
 
-	return param_changed_storage[param_index / bits_per_allocation_unit] &
-	       (1 << param_index % bits_per_allocation_unit);
+	return false;
 }
 
 void param_set_used(param_t param)
 {
-	param_set_used_internal(param);
-}
-
-void param_set_used_internal(param_t param)
-{
-	int param_index = param_get_index(param);
-
-	if (param_index < 0) {
-		return;
+	if (handle_in_range(param)) {
+		// FIXME: this needs locking too
```

Comment:
Can you fix this by letting the bitset use the `atomic` class?

Reply:
Implemented in https://github.com/PX4/Firmware/pull/11318/commits/33a4f8bfa3aa55fba078618277cf629320805460 with new AtomicBitset.